model/src/main/kotlin/com/testerum/model/resources/http/response/xml_body/XmlRootJsObject.kt

```diff
@@ -0,0 +1,94 @@
+package com.testerum.model.resources.http.response.xml_body
+
+import jdk.nashorn.api.scripting.AbstractJSObject
+import org.jdom2.Attribute
+import org.jdom2.Content
+import org.jdom2.Document
+import org.jdom2.Element
+import org.jdom2.Text
+import org.jdom2.input.SAXBuilder
+import org.jdom2.xpath.XPathFactory
+
+const val XPATH_FUNCTION_NAME = "xpath"
+
+class XmlRootJsObject(private val xmlAsUtf8String: String) : AbstractJSObject() {
+    var rootElement: Element? = null
+    var document: Document? = null
+
+    private fun initXmlTree() {
+        if (rootElement != null) {
+            return
+        }
+
+        document = SAXBuilder().build(xmlAsUtf8String.byteInputStream())
```

Comment:
We already have the body as a byte array in ``ValidHttpResponse``. There is no need to transform it into String, just to transform it back into a byte array (this is done in the implementation of ``.byteInputStream()``). Let's just pass the byte array to this class as-is, and pass a ``ByteArrayInputStream(bodyAsBytes)`` to the ``build`` method here.

This avoids making multiple copies of the byte array in memory.